src/goodbase/services/evg_service.py

```diff
@@ -63,15 +64,42 @@ def analyze_build(build: Build) -> Optional[BuildStatus]:
             all_tasks=all_tasks,
         )
 
-    def check_version(self, evg_version: Version, build_checks: List[BuildChecks]) -> bool:
+    def _get_successful_tasks(self, tasks: List[Task], allow_known_failures: bool) -> set[str]:
+        return {
+            task.display_name
+            for task in tasks
+            if task.is_success() or (allow_known_failures and self._is_task_a_known_failure(task))
+        }
+
+    def _is_task_a_known_failure(self, task: Task) -> bool:
+        """
+        Check if a task is a known failure - annotated with an issue or suspected issue that has a BF prefix.
+
+        :param task: the task to check.
+        :return: True if the task is a known failure.
+        """
+        annotations = self.evg_api.get_task_annotation(task.task_id)
+        for annotation in annotations:
+            if annotation.issues:
+                return True
+        return False
+
+    def check_version(
+        self,
+        evg_version: Version,
+        build_checks: List[BuildChecks],
+        allow_known_failures: bool = False,
```

Comment:
Don't forget the docstring for the new parameter :)

Reply:
[nit] I wonder if passing in `False` explicitly is still needed for some of the tests you've fixed below, since it already defaults to `False`?